src/dra/72BB0.c

```diff
@@ -178,7 +178,264 @@ void func_80113148(void) {
+            PLAYER.animFrameDuration = 2;
+            PLAYER.animFrameIdx = 3;
+        }
+    } else if (PLAYER.ext.player.unkAC == 0x65) {
+        func_8010DA48(0x11);
+    }
+    switch (PLAYER.step_s) {
+    case 0x0:
+        if (D_800ACF74 != 0) {
+            D_800ACF74--;
+        } else if (D_80097448[0] >= 0x19) {
+            if (g_Player.unk48 == 0) {
+                x_offset = 0xC;
+                if (PLAYER.facingLeft) {
+                    x_offset = -0xC;
+                }
+                PLAYER.posX.i.hi = x_offset + PLAYER.posX.i.hi;
+                PLAYER.posY.i.hi += 2;
+                func_8011AAFC(g_CurrentEntity, 0xD0004U, 0);
+                D_800ACF74 = 0x60;
+                PLAYER.posY.i.hi -= 2;
+                PLAYER.posX.i.hi -= x_offset;
+            }
+        }
+        local_flags = 6;
+        break;
+    case 0x1:
+        if (!(g_Player.padPressed & PAD_DOWN)) {
+            local_flags = 1;
+            if ((PLAYER.animFrameIdx >= 4U) ||
+                PLAYER.ext.player.unkAC == 0x65) {
+                func_8010DA48(0x13);
+                PLAYER.step_s = 2;
+                PLAYER.animFrameIdx = 2;
+                if (g_Player.unk72 == 1) {
+                    PLAYER.animFrameIdx = 0;
+                }
+                if (g_Player.unk72 == 2) {
+                    PLAYER.animFrameIdx = 1;
+                }
+                return;
+            }
+        } else if (PLAYER.ext.player.unkAC == 0x65) {
+            PLAYER.step_s = 0;
+        } else {
+            if (PLAYER.animFrameDuration < 0) {
+                local_flags = 0x20;
+            }
+        }
+        break;
+    case 0x5:
+        func_8010DFF0(1, 1);
+        if (g_Player.unk72 == PLAYER.animFrameIdx ||
+            PLAYER.animFrameDuration < 0) {
+            local_flags = 0x20;
+        }
+        break;
+    case 0x3:
+    case 0x4:
+        func_8010DFF0(1, 1);
+        if (PLAYER.animFrameDuration < 0) {
+            local_flags = 0x20;
+        }
+        break;
+    case 0x2:
+        local_flags = 1;
+        if (g_Player.padPressed & PAD_DOWN) {
+            func_8010E470(0, PLAYER.velocityX);
+            break;
+        }
+        if (g_Player.unk72 != 0) {
+            func_8010DFF0(1, 1);
+            if (g_Player.unk72 == 1) {
+                PLAYER.animFrameIdx = 0;
+                PLAYER.animFrameDuration = 3;
+                break;
+            } else if (g_Player.unk72 == 2) {
+                if (PLAYER.animFrameIdx != 0) {
+                    PLAYER.animFrameIdx = 1;
+                    PLAYER.animFrameDuration = 3;
+                }
+                break;
+            } else if (g_Player.unk72 == 3) {
+                if ((u16)PLAYER.animFrameIdx >= 2U) {
+                    PLAYER.animFrameIdx = 2;
+                    PLAYER.animFrameDuration = 3;
+                }
+                break;
+            } else if (g_Player.unk72 == 4) {
+                if ((u16)PLAYER.animFrameIdx >= 3U) {
```

Comment:
It looks like some of these casts and Us can be removed

```
            else if(g_Player.unk72 == 4){
                if (PLAYER.animFrameIdx >= 3) {
                    PLAYER.animFrameIdx = 3;
                    PLAYER.animFrameDuration = 3;
                }
                break;
            }
```

Probably the same for other spots